src/Mvc/Mvc.Core/src/DependencyInjection/MvcCoreServiceCollectionExtensions.cs

```diff
@@ -269,7 +273,6 @@ internal static void AddMvcCoreServices(IServiceCollection services)
             //
             // Endpoint Routing / Endpoints
             //
-            services.TryAddSingleton<ActionEndpointDataSource>();
```

Comment:
This should have been removed in the change to removed support for `UseMvc` + `EnableEndpointRouting == true`.